### ISSUE TITLE
Add ability to select where new bookmarks are added

### ISSUE THREAD

**Issue by Nippius** [organicmaps/organicmaps#5512]
**Is your feature request related to a problem? Please describe.**
When adding a new bookmark, the bookmark is added to a (apparently) random bookmark list. Also, as described in issue #5511, the list where the bookmark is added gets shown automatically. If this is a big list (I have a few with hundreds of bookmarks each) it can turn the map unusable until the list is manually hidden.

The problem only gets worse if I need to add several bookmarks in a row (not uncommon).

**Describe the solution you would like**
Either a setting to set the default bookmark list where new bookmarks are saved or the hability to select the bookmark list when creating a new bookmark. Also, as described in #5511 keep hidden list hidden.

**Describe alternatives you have considered**
- How do you solve this issue now with Organic Maps or other apps?
  - The other app I use is google maps and it has a similar problem. In google maps the problem is even worse because the lists are shown without any kind of input. Just wait a few minutes/hours/days and the lists start magically appearing again. Infact, this is exactly the reason why I changed to Organic Maps in the first place. As for how I solve it, I have to hide them manually (the "hide all" button works thankfully) every time.
- Let me know if screenshots/video are needed.

**Additional context**
none

**Comment by biodranik** [comment]
Now bookmarks are automatically added to a _last-used_ bookmarks list. There is already issue when that last-used list is hidden: https://github.com/organicmaps/organicmaps/issues/2201

1. When you say "map is unusable", do you mean it is slow, or it is too polluted by bookmarks?
2. Am I correctly understood that you _do not want_ to add a bookmark to a hidden list? Then you can edit/set the list to a new one, and all future bookmarks will be added there.
3. Can you please test and confirm if selecting a bookmark from a hidden list automatically sets that list as the "last used one" instead of any other recently used list, and all new bookmarks are added into that unhidden list?

**Comment by Nippius** [comment]
Sorry, completely missed #2201 but it is related yes.

1. I meant too polluted. But it's not unmanageable, just a bit annoying but the "Hide All" button solves it.
2. hmm I dont really mind if the list is hidden or not. But I would love to choose where to save it.
2.1 Isn't what you are saying the same as setting a list as the default and save all bookmarks there?
3. It does not. I tried a few more things and it seems that the "last used one" is whatever list a bookmark was added/removed only. For example, let's say the default list is list A. I save a bookmark there and then move it to list B. Afterward, any new bookmarks go to list B. It doesn't matter is list B is hidden or not (either before or after moving the bookmark to it).

But regarding your second question, I can see why this is not easy. For me, if a list hidden and I add a bookmark I expect the list to stay hidden but I can understand that many people will think the other way around makes more sense.

**Comment by pastk** [comment]
- related to #6234

**Comment by pastk** [comment]
> 
>     1. I meant too polluted. But it's not unmanageable, just a bit annoying but the "Hide All" button solves it.

After "Hide All" do you usually have to re-Show the lists you need one-by-one?

Or do you use "Show All"?

**Comment by Nippius** [comment]
> > ```
> > 1. I meant too polluted. But it's not unmanageable, just a bit annoying but the "Hide All" button solves it.
> > ```
> 
> After "Hide All" do you usually have to re-Show the lists you need one-by-one?
> 
> Or do you use "Show All"?

One by one and usually, I only have one list active at any given time. But this is mainly due to the way I organize my lists as I have little, if any, overlap between them.

Actually, now that I think about it, I use the lists more like layers of sorts. For example, as an amateur photographer, I have a list with all the places I want to go take pictures so when I want to go take photos to a certain location, I activate that list and look for the places nearby the selected location. Same thing if I'm looking for a restaurant or a gas station.

**Comment by biodranik** [comment]
Maybe by implementing a way to "archive" unused lists we can solve user issues without changing the behavior of "hide all/show all" buttons?

**Comment by Nippius** [comment]
Like archiving chats/groups in WhatsApp/Signal?  If so, I think that could be a nice workaround. All "archived" lists would be hidden, to show them, just move them from the "archive".

And if the default/last used list is archived but a new place is stored there anyway, a badge or something could be displayed to let the user know where it was stored.

**Comment by biodranik** [comment]
@euf WDYT about "archiving" concept compared to the current hide/show concept?

**Comment by pastk** [comment]
The archiving approach can't replace "hide all / show all" functionality, but could be complimentary to it.
It should help users who have a lot of lists.
However a proper sorting (recently used lists at the top, unused ones in the bottom) kinda does a similar thing.

**Comment by euf** [comment]
I agree, there's a difference between a "hidden" (not shown on a map) and "archived" (not needed for regular interactions and moved away to be used later on never). 

This leads us to a more complex mental model of list management hierarchy. At the moment, the concept is simple, items belong to lists, and all lists are equal. Introducing "archive" would add either add an extra "layer" (lists could be put inside  another abstraction) or have some specific property (which still would need a specific visual cue leading to more abstractions). 

I would not explore this path. We better stick with existing "Places" / "Tracks" inside "Lists" structure. It has only one layer of navigational depth, it's simple to understand, users are already educated to use it, and it is well mapped to the underlying data model from a technical point of view.
